test/test_map_cell.cpp

```diff
@@ -0,0 +1,88 @@
+
+  std::shared_ptr<simulation::Bug> non_empty_bug_1(std::make_shared<simulation::Bug>(nullptr));
+  std::shared_ptr<simulation::Bug> non_empty_bug_2(std::make_shared<simulation::Bug>(nullptr));
+
+  simulation::MapCell::AdjacentCellsReference cells_refs = {
+      std::make_shared<simulation::MapCell>(non_empty_bug_2),
+      std::make_shared<simulation::MapCell>(nullptr),
+      std::make_shared<simulation::MapCell>(non_empty_bug_1),
+      std::make_shared<simulation::MapCell>(nullptr),
+  };
+  simulation::MapCell test_cell = simulation::MapCell(nullptr, cells_refs);
+
+  uint8_t non_empty_bug_count = 0;
+  for(auto cell : test_cell.get_adjacents()) {
+      if(cell->get_bug()) {
+          non_empty_bug_count++;
+      }
+  }
+  EXPECT_EQ(non_empty_bug_count, 2);
+}
+
+TEST(MapCell, GetFreeadjacent_SuccessfullyGetsFirstFreeadjacent) {
+      
+  std::shared_ptr<simulation::Bug> non_empty_bug_1(std::make_shared<simulation::Bug>(nullptr));
+  std::shared_ptr<simulation::Bug> non_empty_bug_2(std::make_shared<simulation::Bug>(nullptr));
+
+  simulation::MapCell::AdjacentCellsReference cells_refs = {
+      std::make_shared<simulation::MapCell>(non_empty_bug_2),
+      std::make_shared<simulation::MapCell>(nullptr),
+      std::make_shared<simulation::MapCell>(non_empty_bug_1),
+      std::make_shared<simulation::MapCell>(nullptr),
+  };
+  simulation::MapCell test_cell = simulation::MapCell(nullptr, cells_refs);
+
+  EXPECT_EQ(test_cell.get_free_adjacent(), cells_refs.down);
+}
+
+TEST(MapCell, GetFreeadjacent_GetsNullReferenceIfNoneEmpty) {      
+  std::shared_ptr<simulation::Bug> non_empty_bug_1(std::make_shared<simulation::Bug>(nullptr));
+  std::shared_ptr<simulation::Bug> non_empty_bug_2(std::make_shared<simulation::Bug>(nullptr));
+
+  simulation::MapCell::AdjacentCellsReference cells_refs = {
+      std::make_shared<simulation::MapCell>(non_empty_bug_2),
+      std::make_shared<simulation::MapCell>(non_empty_bug_2),
+      std::make_shared<simulation::MapCell>(non_empty_bug_1),
+      std::make_shared<simulation::MapCell>(non_empty_bug_1),
+  };
+  simulation::MapCell test_cell = simulation::MapCell(nullptr, cells_refs);
+
+  EXPECT_EQ(test_cell.get_free_adjacent(), nullptr);
+}
+
+TEST(MapCell, MoveBug_BugMovedSuccessfully) {
+  std::shared_ptr<simulation::Bug> non_empty_bug_1(std::make_shared<simulation::Bug>(nullptr));
+  std::shared_ptr<simulation::Bug> non_empty_bug_2(std::make_shared<simulation::Bug>(nullptr));
+
+  simulation::MapCell::AdjacentCellsReference cells_refs = {
+      std::make_shared<simulation::MapCell>(non_empty_bug_2),
+      std::make_shared<simulation::MapCell>(nullptr),
+      std::make_shared<simulation::MapCell>(nullptr),
+      std::make_shared<simulation::MapCell>(nullptr),
+  };
+  simulation::MapCell test_cell = simulation::MapCell(non_empty_bug_1, cells_refs);
+  test_cell.move_bug(cells_refs.left);
+  EXPECT_EQ(test_cell.get_bug(), nullptr);
+  EXPECT_EQ(cells_refs.left->get_bug(), non_empty_bug_1);
```

Comment:
Same case as before. How about leaving `cells_refs` with only one free cell so it is evident where the bug will move to?